aptos-move/aptos-gas/src/transaction/storage.rs

```diff
@@ -48,49 +48,32 @@ impl StoragePricingV1 {
             }
     }
 
-    fn calculate_write_set_gas<'a>(
-        &self,
-        ops: impl IntoIterator<Item = (&'a StateKey, &'a WriteOp)>,
-    ) -> InternalGas {
-        use WriteOp::*;
-
-        // Counting
-        let mut num_ops = NumArgs::zero();
-        let mut num_new_items = NumArgs::zero();
-        let mut num_bytes_key = NumBytes::zero();
-        let mut num_bytes_val = NumBytes::zero();
-
-        for (key, op) in ops {
-            num_ops += 1.into();
-
-            if self.write_data_per_byte_in_key > 0.into() {
-                // TODO(Gas): Are we supposed to panic here?
-                num_bytes_key += NumBytes::new(
+    fn io_gas_per_write(&self, key: &StateKey, op: &WriteOp) -> InternalGas {
```

Comment:
@msmouse I had to rewrite these cost functions so to allow gas meters to be composable. This is not supposed to have any effect on the actual formula, but it's something we'll need to verify via replays.